test-data/unit/check-based-type-render.test

```diff
@@ -14,6 +14,17 @@ class A(Generic[T2]):
 
 reveal_type(A.f)  # N: Revealed type is "def [T2 (from A), T] (self: __main__.A[T2], t: T, t2: T2) -> T | T2"
 reveal_type(A[int]().f)  # N: Revealed type is "def [T] (t: T, t2: int) -> T | int"
+A.f = 1  # E: Cannot assign to a method  [assignment] \
+         # E: Incompatible types in assignment (expression has type "int", variable has type "[T2 (from A), T] (A[T2], T, T2) -> T | T2")  [assignment]
```

Comment:
> T2 (from A)

what does it look like if `T2` has a bound?

Reply:
`def [T (from A): int, T2: str] (self: test.A[T], t2: T2) -> None`

Reply:
is there a test for it

Reply:
done